gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java

```diff
@@ -229,6 +237,55 @@ public void updateEntity(
     }
   }
 
+  @Override
+  public void getBaselineForEntities(
+      BaselineEntitiesRequest request, StreamObserver<BaselineEntitiesResponse> responseObserver) {
+    Optional<String> tenantId =
+        org.hypertrace.core.grpcutils.context.RequestContext.CURRENT.get().getTenantId();
+    if (tenantId.isEmpty()) {
+      responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
+      return;
+    }
+
+    try {
```

Comment:
Maybe move validations to a class or method of their own like https://github.com/hypertrace/gateway-service/blob/main/gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/validators/request/RequestValidator.java

Reply:
Fixed and added unitests